Source/WebCore/platform/graphics/cocoa/WebCoreDecompressionSession.mm

```diff
@@ -153,8 +154,38 @@ static bool canCopyFormatDescriptionExtension()
         auto result = VTDecompressionSessionCreate(kCFAllocatorDefault, videoFormatDescription, (__bridge CFDictionaryRef)videoDecoderSpecification, (__bridge CFDictionaryRef)m_pixelBufferAttributes.get(), nullptr, &decompressionSessionOut);
         if (noErr == result) {
             m_decompressionSession = adoptCF(decompressionSessionOut);
-            CFArrayRef rawSuggestedQualityOfServiceTiers = nullptr;
-            VTSessionCopyProperty(decompressionSessionOut, kVTDecompressionPropertyKey_SuggestedQualityOfServiceTiers, kCFAllocatorDefault, &rawSuggestedQualityOfServiceTiers);
+#if PLATFORM(VISION)
+            m_stereoViewMap.clear();
+            RetainPtr<CFMutableArrayRef> layersArray = nullptr;
+            // Get the tag collection array
+            CFArrayRef rawTagCollectionArray = nullptr;
+            if (auto status = PAL::CMVideoFormatDescriptionCopyTagCollectionArray(videoFormatDescription, &rawTagCollectionArray); status == noErr && rawTagCollectionArray) {
```

Comment:
Nit: this is a pretty deep nested `if/for/if/if` section. Consider refactoring to use `continue` or early `return` to flatten and make the resulting code more readable.

Reply:
it's a draft still.